### PR TITLE
fix(ci_visibility): support pytest 8.0.0

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
Addresses https://github.com/DataDog/dd-trace-py/issues/8220 and fixes an issue with `pytest` `8.x` and above 
(brought by https://github.com/pytest-dev/pytest/issues/11137 ) where `pytest.Package` objects no longer have an attached `module` attribute.

This also changes the testing matrix to include version `~=8.0`, but maintains `~=7.0` as a separate testing environment.

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/contrib/pytest/test_pytest.py [hunk 1]
```diff
@@ -3435,7 +3565,7 @@ def multiply_two_number_list(list_1, list_2):
                         """
                     from tools import add_two_number_list
 
-                    def test_add_two_number_list():
+                    def test_add_two_number_lis7418t():
```
Comment (romainkomorndatadog) [inline tests/contrib/pytest/test_pytest.py:3568]:
```suggestion
                    def test_add_two_number_list():
```

**Comment by romainkomorndatadog** [comment]
The weeds are getting a bit deeper. I need to rewrite everything that touches `item.parent` because we now have items where the parent is no longer what it used to be (namely, there are `pytest.Dir` type objects where there used to be `pytest.Package` objects).

We have long-delayed plans to introduce an API to centralize all the span/event creation into the `CIVisibility` service, and greatly simplify the integration-side code, and I've made the decision to do that work now, instead of making a slew of changes in our plugin that I'll unwind later this quarter.

That does mean slowing down the response to #8220 , but it should be better for the long term.

**Review by juan-fernandez — APPROVED** [review]

File: releasenotes/notes/ci_visibility-fix-pytest-8.0.0-support-88fcb7b5144d7efa.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  - |
+    For reporting bug fixes.
+    Use past tense for the problem and present tense for the fix and solution in the following format:
+    Format: <scope>: This fix resolves an issue where <ABC_bug> caused <XYZ_situation>.
+other:
+  - |
+    For any change which does not fall into any of the above categories. Since changes falling into this category are 
+    likely rare and not very similar to each other, no specific format other than a required scope is provided. 
+    The author is requested to use their best judgment to ensure a quality release note.
+    Format: <scope>: <add_release_note_here>.
```
Comment (mabdinur) [inline releasenotes/notes/ci_visibility-fix-pytest-8.0.0-support-88fcb7b5144d7efa.yaml:65]:
```suggestion
features:
  - |
   pytest: Adds support for v8.0
```

**Review by mabdinur — APPROVED** [review]
We'll need to update the release note. Aside from that LGTM

**Pull request merged**